src/type/definition.js

```diff
@@ -782,6 +853,16 @@ export type GraphQLObjectTypeConfig<TSource, TContext> = {
   extensionASTNodes?: ?$ReadOnlyArray<ObjectTypeExtensionNode>,
 };
 
+export type GraphQLObjectTypeExactConfig<TSource, TContext> = {|
+  name: string,
+  interfaces: Array<GraphQLInterfaceType>,
+  fields: GraphQLFieldExactConfigMap<TSource, TContext>,
+  isTypeOf: ?GraphQLIsTypeOfFn<TSource, TContext>,
+  description: ?string,
+  astNode?: ?ObjectTypeDefinitionNode,
+  extensionASTNodes: $ReadOnlyArray<ObjectTypeExtensionNode>,
+|};
```

Comment:
I can't use `GraphQL*Config` types since they have `Thunk` fields and a lot of fields marked as optional so without this new type code that uses `toConfig` should have a lot of unnecessary runtime checks and also have dead code to handle `Thunk`.

Plus I think it easier to keep this types up to date than update `extendSchema`, `lexicographicSortSchema`, and all similar functions inside 3rd-party tools.

Reply:
I wonder whether we still need the `Thunk` fields in the `GraphQL*Config` types at all. This feels like a much cleaner expression of a GraphQL type without needing surrounding schema. I'll admit, I don't fully understand how the thunks are used currently, so take that idea as a relative newcomer's curiosity instead of a strong suggestion.

Reply:
As far as I have always understood, the thunks are there to avoid circular references at module import time when your schema is made up of types each in their own module. We have definitely run into that situation a couple of times and switched to using thunks.

For `toConfig`, however, it’s indeed not necessary as by the time it can be invoked the import stage should already have passed.